topeft/modules/datacard_tools.py

```diff
@@ -787,7 +787,14 @@ def analyze(self,km_dist,ch,selected_wcs, crop_negative_bins):
         outf_root_name = self.FNAME_TEMPLATE.format(cat=ch,kmvar=km_dist,ext="root")
 
         h = self.hists[km_dist]
+        try:
+            h_sumw2 = self.hists[km_dist+"_sumw2"]
+        except KeyError:
+            msg = "No sumw2 histogram found! Setting errors to 0"
+            print(msg)
+            h_sumw2 = None
```

Comment:
why a `try/except` clause here instead of an `if`? I think this could be rewritten like so:
```
h_sumw2 = None
if f"{km_dist}_sumw2" in self.hists:
  h_sumw2
```
and if you still wanted to have the warning message printed, you could add an `else` block for that situation. If you wanted to be _really_ compact you could rewrite it in one line as:
```
h_sumw2 = self.hists[km_dist+"_sumw2"] if f"{km_dist}_sumw2" in self.hists else None
```
But I'm not the biggest fan of using the ternary conditional in code, unless it's for _really_ simple checks and I think this particular case is borderline too complex

Reply:
Ok, I've switched to an `if/else` blocok.